docs/index-type-operator.md

```diff
@@ -0,0 +1,88 @@
+# `index` type operators
+
+## Summary
+
+This RFC proposes the addition of one type operator, `index`, which can be used to look up a specific property of another type (like TypeScript's Indexed Access Type).
+
+## Motivation
+
+The primary motivation of this proposal is to allow Luau to define a type by accessing a property of another existing type. For instance, consider the following example code:
+```lua
+type Person = {
+  age: number,
+  name: string,
+  alive: boolean
+}
+
+local bob: Person = {
+  age = 22,
+  name = "Bob",
+  alive = true
+}
+
+local function doSmt(param: typeof(bob["age"])) -- param = string
```

Comment:
found a smol typo: age should be a number here, not a string (as defined by the Person type on line 11)
```suggestion
local function doSmt(param: typeof(bob["age"])) -- param = number
```

Reply:
oops, thank you for catching this!